core/components/ui/header/locale-switcher.tsx

```diff
@@ -55,6 +53,18 @@ const LocaleSwitcher = ({ activeLocale, locales }: Props) => {
     [locales],
   );
 
+  useEffect(() => {
+    if (regionSelected && languageSelected) {
+      const nextLocale = locales.find(
+        (locale) => locale.language === languageSelected && locale.region === regionSelected,
+      );
+
+      if (nextLocale) {
+        setNewLocale(nextLocale.id);
+      }
+    }
+  }, [regionSelected, languageSelected, locales]);
```

Comment:
Can we compute `newLocale` instead of this effect? Or use a memo?

Reply:
Sure!